st2common/st2common/util/green/shell.py

```diff
@@ -80,29 +80,30 @@ def run_command(cmd, stdin=None, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
     process = subprocess.Popen(args=cmd, stdin=stdin, stdout=stdout, stderr=stderr,
                                env=env, cwd=cwd, shell=shell, preexec_fn=preexec_func)
 
-    def on_timeout_expired(timeout):
-        global timed_out
-
-        try:
-            process.wait(timeout=timeout)
-        except subprocess.TimeoutExpired:
-            # Command has timed out, kill the process and propagate the error.
-            # Note: We explicitly set the returncode to indicate the timeout.
-            process.returncode = TIMEOUT_EXIT_CODE
-
-            if kill_func:
-                kill_func(process=process)
-            else:
-                process.kill()
-
-    timeout_thread = eventlet.spawn(on_timeout_expired, timeout)
-    stdout, stderr = process.communicate()
-    timeout_thread.cancel()
-    exit_code = process.returncode
-
-    if exit_code == TIMEOUT_EXIT_CODE:
-        timed_out = True
+    if not timeout:
+        stdout, stderr = process.communicate()
```

Comment:
Isn't this dangerous? We should pretty much not support unlimited timeout. Wouldn't this block the worker?